test/rules/resources/test_deletionpolicy.py

```diff
@@ -0,0 +1,34 @@
+"""
+  Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  software and associated documentation files (the "Software"), to deal in the Software
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+from cfnlint.rules.resources.DeletionPolicy import DeletionPolicy  # pylint: disable=E0401
+from .. import BaseRuleTestCase
+
+
+class TestResourceDeletionPolicy(BaseRuleTestCase):
+    """Test base template"""
+    def setUp(self):
+        """Setup"""
+        super(TestResourceDeletionPolicy, self).setUp()
+        self.collection.register(DeletionPolicy())
+
+    def test_file_positive(self):
```

Comment:
I think generally, these tests need to be more explicit. What failure case is expected here? There are several possible; 
* That the DeletionPolicy is an array and not a string? 
* That the DeletionPolicy value is not in the enum
* That `Snapshot` is applied to an invalid type.

Will catch more faults and make tests clearer if we target them a bit more explicitly.

Reply:
So ...

1. This is a "Chuck is a crap dev moment" -- the DeletionPolicy code from #631 ended up here as well.

2. I test the three explicit scenarios via the new bad template.  That is historically how we have tested rules. Are we proposing a new, non-template mechanism? Or that I should break the three bad cases into separate templates?

Reply:
Ok, I've narrowed this just the changes for UpdateReplacePolicy.  Now help me understand what improvements you're looking for with testing?